src/WebJobs.Extensions.SendGrid/Bindings/SendGridHelpers.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Net.Mail;
+using Microsoft.Azure.WebJobs.Extensions.SendGrid;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using SendGrid.Helpers.Mail;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Bindings
+{
+    internal class SendGridHelpers
+    {
+        internal static EmailAddress Apply(EmailAddress current, string value)
+        {
+            EmailAddress mail;
+            if (TryParseAddress(value, out mail))
+            {
+                return mail;
+            }
+            return current;
+        }
+
+        internal static bool TryParseAddress(string value, out EmailAddress email)
+        {
+            email = null;
+
+            if (string.IsNullOrEmpty(value))
+            {
+                return false;
+            }
+
+            try
+            {
+                // MailAddress will auto-parse the name from a string like "testuser@test.com <Test User>"
+                MailAddress mailAddress = new MailAddress(value);
+                string displayName = string.IsNullOrEmpty(mailAddress.DisplayName) ? null : mailAddress.DisplayName;
+                email = new EmailAddress(mailAddress.Address, displayName);
+                return true;
+            }
+            catch (FormatException)
+            {
+                return false;
+            }
+        }
+
+        internal static void DefaultMessageProperties(SendGridMessage mail, SendGridConfiguration config, SendGridAttribute attribute)
+        {
+            // Apply message defaulting
+            if (mail.From == null)
+            {
+                if (!string.IsNullOrEmpty(attribute.From))
+                {
+                    EmailAddress from = null;
+                    if (!TryParseAddress(attribute.From, out from))
+                    {
+                        throw new ArgumentException("Invalid 'From' address specified");
+                    }
+                    mail.From = from;
+                }
+                else if (config.FromAddress != null)
+                {
+                    mail.From = config.FromAddress;
+                }
+            }
+
+            if (!IsToValid(mail))
```

Comment:
a lot of this looks like it could be simplified with something like the Twilio Utility.FirstOrDefault() that does all the null/empty coalescing

Reply:
Agreed. However my primary focus in these migrations is to structurally migrate w/o too much uneccessary refactoring. We can improve later :)

Reply:
Agreed, sounds good